### DISCUSSION TITLE
Can a Text in a layout auto scroll-up like a sub-console ?

### DISCUSSION THREAD

**Discussion by treejiang** [Textualize/rich#1181] [Q&A] [answered]
I'm developing a full-screen TUI application. 
After I use the layout method, I add a Text into a layout, or through a Panel and use the "with live layout", everything is OK.

But when I try to continuously Text.append or Table.add_row, when the vertical lines is over the Box, the content is cut and can't display by auto scrolling up like a common console.

Can some function or attribute to achieve it ? ( I've tried vertical_overflow="visible" in Live, but it doesn't work).

Thank you very much.

**Comment by willmcgugan** [comment, marked as answer]
Rich doesn't currently have much in the way of TUI support. At some point there will be a scrolling "window" which sounds like what you are looking for.

**Reply by treejiang** [reply]
Got it.  Yes, a auto scrolling "window" component is what I'm looking for.  Text in Pannel seems like it, and it works very well on static or updating style.  Maybe I write some funcs like counting lines or managing text buffers etc, can work with the update/refresh method and achieve the funciton what I want.

Anyway,  thank you very much for your reply.
